### PR TITLE
DOCSP-43829 - Disposable IMongoClient

### PR DESCRIPTION
**Pull request by mongoKart**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-43829

### Staging Links

<li><a href=https://deploy-preview-241--mongodb-docs-csharp.netlify.app/whats-new>whats-new</a></li>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/whats-new.txt [hunk 1]
```diff
@@ -57,7 +58,14 @@ The 3.0 driver release includes the following new features:
   This type is available in .NET 5 and later.
   To learn more about the ``Half`` type, see the
   `Half Struct <https://learn.microsoft.com/en-us/dotnet/api/system.half?view=net-8.0>`__
-  API reference page on MSDN. 
+  API reference page on MSDN.
+
+- The ``IMongoClient`` interface inherits the ``IDisposable`` interface. As a result, the
```
Comment (mongoKart) [inline source/whats-new.txt:63]: Tech reviewer: I don't see this as a breaking change, but correct me if I'm wrong

File: source/whats-new.txt [hunk 2]
```diff
@@ -57,7 +58,14 @@ The 3.0 driver release includes the following new features:
...
+  ``MongoClient`` class and other classes that implement the ``IMongoClient`` interface
+  contain a ``Dispose()`` method. To learn more about this interface and use of the
+  ``Dispose()`` method, see
+  `Dispose Pattern <https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/dispose-pattern>`__
+  on MSDN.
```
Comment (rustagir) [inline source/whats-new.txt:68]:
```suggestion
  contain a ``Dispose()`` method. To learn more about the ``IDisposable`` interface and how to use the
  ``Dispose()`` method, see
  `Dispose Pattern <https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/dispose-pattern>`__
  on MSDN.
```

**Review by rustagir — APPROVED** [review]
lgtm with a nit suggestion

File: source/whats-new.txt (hunk 1 above)
Comment (BorisDog) [reply source/whats-new.txt:63]: Yes, this is not a breaking change.
We should mention that `Dispose` disposes the underlying cluster and  connections to the server.
Also this is an experimental feature, so we should document it as such.

**Review by BorisDog — CHANGES_REQUESTED** [review]
Left a comment

**Review by BorisDog — APPROVED** [review]
LGTM

**Pull request merged**